### PR TITLE
route notifications to relevant channel based on config file

### PR DESCRIPTION
**Pull request by ironcamel**

### CODE REVIEW

File: config.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+
+channel_map:
+  'chef': '#sysops'
+  'crowdtilt-internal-api': '#engineering'
+  'crowdtilt-public-site': '#engineering'
```
Comment (tildedave) [inline config.yml]: `'tilt-images': '#engineering'`

**Comment by tildedave** [comment]
What do you think about putting the style guides in here too?  I think engineering is the best place for those to go.

**Comment by ironcamel** [comment]
+1 on style guides

File: server.rb [hunk 2]
```diff
@@ -74,6 +75,10 @@ def comment_created_message(redis, repository, issue, comment)
   end
 end
 
+def _channel(event)
```
Comment (cohitre) [inline server.rb]: rename to `slack_channel_id()`

File: lib/slack_api.rb [hunk 3]
```diff
@@ -12,6 +12,7 @@ def list_channels
   end
 
   def post_message(channel_id, message, options={})
+    return unless channel_id
```
Comment (cohitre) [inline lib/slack_api.rb]: i added a `valid_event?` method to master. we should check if the channel_id is present there. ideally by the time we get to post_message we are committed to processing the event.

**Comment by cohitre** [comment]
i left a couple must fix comments. it lgtm besides that

**Pull request merged**